RecoVertex/BeamSpotProducer/src/PVFitter.cc

```diff
@@ -318,9 +320,14 @@ bool PVFitter::runFitter() {
 
     if ( pvStore_.size() <= minNrVertices_ ) return false;
 
-    TH1F *h1PVx = (TH1F*) hPVx->ProjectionX("h1PVx", 0, -1, "e");
-    TH1F *h1PVy = (TH1F*) hPVy->ProjectionX("h1PVy", 0, -1, "e");
-    TH1F *h1PVz = (TH1F*) hPVx->ProjectionY("h1PVz", 0, -1, "e");
+    //need to create a unique histogram name to avoid ROOT trying to re-use one
+    // also tell ROOT to hide its global state
+    TDirectory::TContext guard{nullptr};
```

Comment:
@Dr15Jones : mainly for my own education, could you please explain what this line is meant to? Thank you!
Also the class documentation in root doesn't look very explanatory to me:
https://root.cern.ch/doc/master/classTDirectory_1_1TContext.html

Reply:
This temporarily sets the global `gDirectory` pointer to `nullptr` which avoids accidental sharing across threads. When `TDirectory::TContext` is deleted, it rests `gDirectory` back to what it was before the constructor was called.